apps/src/weblab/brambleHost.js

```diff
@@ -52,28 +52,58 @@ function putFilesInBramble(sources, callback) {
     // create project root directory
     sh.mkdirp(projectRoot, err => {
 
-      function writeFileData(path, data, callback) {
-        path = Path.join(projectRoot, path);
+      function writeFileDataAndContinue(filename, data, currentIndex) {
```

Comment:
This function no longer takes an arbitrary callback - it is now specific to the scenario and knows to call `writeSourceFile` on the next index when complete. Hence, the rename to `writeFileDataAndContinue`.